### PR TITLE
State Entity class

### LINKED ISSUE #13: State Machine (State Entity)

### PR DESCRIPTION
**Pull request by NoePolytech**
closes #13

### CODE REVIEW

File: Engine/Entities/StateEntity.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            State.OnStateStart(this);
+        }
+
+        /// <summary>
+        /// Updates the state of the entity.
+        /// </summary>
+        public override void Update()
+        {
+            State.OnStateUpdate(this);
+        }
```
Comment (jumsee) [inline Engine/Entities/StateEntity.cs:41]: The documentation should be more explicit to differentiate between these two methods.

File: Engine/Entities/States/IState.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+{
+    /// <summary>
+    /// Represents what can a StateEntity should do in a specific situation.
+    /// </summary>
+    public interface IState
+    {
+        /// <summary>
+        /// Called when the state has just been assigned on an entity.
+        /// </summary>
+        /// <param name="stateEntity">The entity that just started to be in this state.</param>
```
Comment (jumsee) [inline Engine/Entities/States/IState.cs:11]:
```suggestion
        /// <param name="stateEntity">The entity that transitioned to this state.</param>
```

File: Engine/Entities/StateEntity.cs (hunk 1 above)
Comment (NoePolytech) [reply Engine/Entities/StateEntity.cs:41]: On Update and OnStateUpdate methods ?

Comment (jumsee) [reply Engine/Entities/StateEntity.cs:41]: ChangeState and Update (the two methods I included in the review...)

Comment (NoePolytech) [reply Engine/Entities/StateEntity.cs:41]: 🆗

**Pull request merged**